### DISCUSSION TITLE
Gotta be a better way to debug cloud-config…

### DISCUSSION THREAD

**Discussion by fullmetal-fred** [kairos-io/kairos#447] [Q&A] [answered]
Hey all! Any tips for debugging cloud-configs? Trial by fire? Is there a linter? Syntax errors can be hard to catch sometimes and not entirely sure of best practices here. For example, where are logs stored for what happened during each stage?

Thanks!

**Comment by mudler** [comment, marked as answer]
Hey! 

good point, we need to both document those locations and  until we get a proper validation mechanism can just give some tips. I'm currently collecting them in the docs page, feel free to sneak peek until it's ready and feel free to give any feedback: https://github.com/kairos-io/kairos/pull/448 , appreciated!

**Reply by fullmetal-fred** [reply]
Brilliant, it already looks amazing! For others: https://kairos.io/docs/reference/configuration/#test-your-cloud-configs
Great work @mudler!

**Reply by sdwilsh** [reply]
Before I found this discussion, I was looking for steps to figure this out in https://kairos.io/docs/reference/troubleshooting/.  It might be worthwhile to add a link to https://kairos.io/docs/reference/configuration/#test-your-cloud-configs on that page to help folks out.
